tests/schema/tests.py

```diff
@@ -3191,3 +3191,164 @@ class Meta:
         with connection.schema_editor(atomic=True) as editor:
             editor.alter_db_table(Foo, Foo._meta.db_table, 'renamed_table')
         Foo._meta.db_table = 'renamed_table'
+
+    @isolate_apps('schema')
+    @skipUnlessDBFeature('supports_collation_on_charfield')
+    def test_collations(self):
+        if connection.vendor == 'mysql':
+            collation = connection.features.test_collations['swedish-ci']
+        else:
+            collation = list(connection.features.test_collations.values())[0]
+
+        class CollationTest(Model):
+            field = CharField(max_length=255, db_collation=collation)
+
+            class Meta:
+                app_label = 'schema'
+
+        self.isolated_local_models = [CollationTest]
+        with connection.schema_editor() as editor:
+            editor.create_model(CollationTest)
+
+        with connection.cursor() as cursor:
+            table_description = connection.introspection.get_table_description(cursor, CollationTest._meta.db_table)
+        self.assertEqual(table_description[1].collation, collation)
+
+        field_without_collation = CharField(max_length=255, db_collation=None)
+        field_without_collation.set_attributes_from_name('field')
+        with connection.schema_editor() as editor:
+            editor.alter_field(
+                CollationTest, CollationTest._meta.get_field('field'), field_without_collation, strict=True
+            )
+        with connection.cursor() as cursor:
+            table_description = connection.introspection.get_table_description(cursor, CollationTest._meta.db_table)
+        if connection.vendor == 'mysql':
+            # MySQL gives the collation of the table if there's no collation set.
+            with connection.cursor() as cursor:
+                table_collation = connection.introspection.get_table_collation(cursor, CollationTest._meta.db_table)
+            self.assertEqual(table_description[1].collation, table_collation)
+        elif connection.vendor == 'oracle':
+            self.assertEqual(table_description[1].collation, 'USING_NLS_COMP')
```

Comment:
Can we add `'default'` to `test_collations`?

Reply:
I've changed it to be more like the MySQL test above, which I think is a bit more robust. I don't know if there's a good reason to add `'default'` there as the only place to use it is here, and every single vendor is a special case, so it wouldn't really be used.